test/e2e/upgrade/upgrade.go

```diff
@@ -569,10 +596,10 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
 			// record whether the cluster was fast or slow upgrading.  Don't fail the test, we still want signal on the actual tests themselves.
 			upgradeEnded := time.Now()
 			upgradeDuration := upgradeEnded.Sub(upgradeStarted)
-			testCaseName := fmt.Sprintf("[sig-cluster-lifecycle] cluster upgrade should complete in %0.2f minutes", durationToSoftFailure.Minutes())
+			testCaseName := fmt.Sprintf("[sig-cluster-lifecycle] cluster upgrade should complete in a reasonable time")
 			failure := ""
-			if upgradeDuration > durationToSoftFailure {
-				failure = fmt.Sprintf("%s to %s took too long: %0.2f minutes", action, versionString(desired), upgradeDuration.Minutes())
+			if upgradeDuration > upgradeDurationLimit {
+				failure = fmt.Sprintf("%s to %s took too long: %0.2f minutes (for this platform/network, it should be less than %0.2f minutes)", action, versionString(desired), upgradeDuration.Minutes(), upgradeDurationLimit.Minutes())
```

Comment:
Just a side note you could consider, if the test duration you already looked at isn't sufficient, extracting test metadata here on import and then a grafana panel would be a pretty quick path to charting upgrade times by day.

Reply:
We have the by-day view in Sippy https://sippy.dptools.openshift.org/sippy-ng/tests/4.14/analysis?test=Cluster%20upgrade.%5Bsig-cluster-lifecycle%5D%20Cluster%20completes%20upgrade

If we wire up grafana to postgres we should be able to see it without any extra data collection